docker-rancher/server.json

```diff
@@ -14,135 +15,95 @@
-        "Standard_A9",
-        "Standard_A10",
-        "Standard_A11"
+        "16.04-LTS",
+        "18.04-LTS"
       ],
-      "defaultValue": "Standard_A2"
+      "type": "String",
+      "metadata": {
+        "description": "Set the OS Image SKU"
+      }
     },
-    "rancherCount": {
-      "type": "int",
+    "storageAccountType": {
+      "defaultValue": "Standard_LRS",
+      "allowedValues": [
+        "Standard_LRS",
+        "Premium_LRS"
+      ],
+      "type": "String",
       "metadata": {
-        "description": "Will be ignored, solely for integration purposes"
-      },
-      "minValue": 1,
-      "maxValue": 1
+        "description": "Set the Virtual Machine OS Disk type"
+      }
     },
     "location": {
       "type": "string",
-      "defaultValue": "[resourceGroup().location]",
       "metadata": {
-        "description": "Location for all resources."
+        "description": "Location for all resources, ex eastus2, westus2..."
       }
     }
   },
   "variables": {
-    "rancherCount": 1,
-    "availabilitySet": "[parameters('rancherNodeName')]",
-    "storageAccountName": "[concat(uniquestring(resourceGroup().id), 'rancherhost')]",
-    "networkSecurityGroupName": "[parameters('rancherNodeName')]",
-    "imagePublisher": "Canonical",
-    "imageOffer": "UbuntuServer",
-    "ubuntuOSVersion": "14.04.5-LTS",
-    "virtualNetworkName": "[parameters('rancherNodeName')]",
+    "availabilitySetName": "[toLower(concat(parameters('rancherServerName'), '-avset'))]",
+    "vmSize": "[parameters('rancherVmSize')]",
+    "osImagePublisher": "Canonical",
+    "osImageOffer": "UbuntuServer",
+    "osImageVersion": "latest",
+    "virtualNetworkName": "[toLower(concat(parameters('rancherServerName'), '-vnet'))]",
     "addressPrefix": "192.168.254.0/24",
-    "subnetName": "[parameters('rancherNodeName')]",
+    "subnetName": "[toLower(concat(parameters('rancherServerName'), '-subnet'))]",
     "subnetPrefix": "192.168.254.0/24",
+    "subnetRef": "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworkName'), variables('subnetName'))]",
+    "NICPrefix": "-nic",
     "publicIPAddressType": "Dynamic",
-    "storageAccountType": "Standard_LRS",
-    "vmStorageAccountContainerName": "vhds",
-    "vmSize": "[parameters('rancherVmSize')]",
-    "extensionName": "DockerExtension",
-    "subnetRef": "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworkName'), variables('subnetName'))]"
+    "publicIPAddressName1": "[toLower(concat(parameters('rancherServerName'), '-pip1'))]",
+    "networkSecurityGroupName": "[toLower(concat(parameters('rancherServerName'), '-nsg'))]",
+    "extensionName": "DockerExtension"
   },
   "resources": [
     {
-      "type": "Microsoft.Storage/storageAccounts",
-      "name": "[variables('storageAccountName')]",
-      "apiVersion": "2015-06-15",
+      "type": "Microsoft.Compute/availabilitySets",
+      "apiVersion": "2017-03-30",
```

Comment:
update these when you remove the profile above (won't comment on the rest, CI will catch them)

Reply:
ok